### PR TITLE
Obsolete performance counter features

### PR DESCRIPTION
**Pull request by kbaley**
Mark CriticalTime and SLA features obsolete

### CODE REVIEW

**Comment by kbaley** [comment]
Marked as WIP because I had some trouble getting the acceptance tests project to run due to the now-obsolete features. Even after adding the "suppress warnings" setting. Waiting to see if this passes on the build server.

**Comment by timbussmann** [comment]
@kbaley have a look at this example tests which disables the 618 obsolete warning for the test: https://github.com/Particular/NServiceBus/blob/e0b26040e06db6bffa2595df0de0c5aeeb0996eb/src/NServiceBus.AcceptanceTests/Core/Encryption/When_using_Rijndael_with_multikey.cs#L2
(and don't forget to reenable it https://github.com/Particular/NServiceBus/blob/e0b26040e06db6bffa2595df0de0c5aeeb0996eb/src/NServiceBus.AcceptanceTests/Core/Encryption/When_using_Rijndael_with_multikey.cs#L83)

**Comment by kbaley** [comment]
@timbussmann Okay thanks. I did do that locally to make sure the tests all passed. Didn't realize it was okay to add the pragma warnings in.

**Comment by kbaley** [comment]
@timbussmann Any idea what adding the code to the projects Suppress Warnings list doesn't work?

File: src/NServiceBus.Core/Performance/Statistics/CriticalTime/CriticalTimeMonitoringConfig.cs [hunk 1]
```diff
@@ -5,6 +5,10 @@ namespace NServiceBus
     /// <summary>
     /// Provide configuration options for monitoring related settings.
     /// </summary>
+    [ObsoleteEx(
+        Message = "Performance counters have been released as a separate package: NServiceBus.WindowsPerformanceCounters",
+        RemoveInVersion = "8",
+        TreatAsErrorFromVersion = "7")]
     public static class CriticalTimeMonitoringConfig
```
Comment (timbussmann) [inline src/NServiceBus.Core/Performance/Statistics/CriticalTime/CriticalTimeMonitoringConfig.cs]: you should also obsolete the extension method I think as you'll probably never reference this class directly. Have you tested this?

File: src/NServiceBus.Msmq.AcceptanceTests/NServiceBus.Msmq.AcceptanceTests.csproj [hunk 2]
```diff
@@ -20,6 +20,8 @@
     <DefineConstants>DEBUG;TRACE</DefineConstants>
     <ErrorReport>prompt</ErrorReport>
     <WarningLevel>4</WarningLevel>
+    <NoWarn>618</NoWarn>
```
Comment (timbussmann) [inline src/NServiceBus.Msmq.AcceptanceTests/NServiceBus.Msmq.AcceptanceTests.csproj]: should be removed

**Comment by kbaley** [comment]
@timbussmann Updated. I had the 618 warning suppressed in the wrong project which is why it wasn't working. But I stuck with the pragma calls as per your suggestion.

**Comment by timbussmann** [comment]
@kbaley doesn't `ReceiveStatisticsPerformanceCounters` need to be obsoleted as well?

**Comment by kbaley** [comment]
@timbussmann That class wasn't moved over.

@dvdstelt I don't recall if we discussed the `ReceiveStatisticsPerformanceCounters` feature. Is it not required?

**Comment by timbussmann** [comment]
@kbaley @dvdstelt I'm good to pull this in regardless but I guess you still want to double check on `ReceiveStatisticsPerformanceCounters`.

**Comment by dvdstelt** [comment]
@timbussmann @kbaley GOOD CALL on the `ReceiveStatisticsPerformanceCounters`, how did you find that?
Because if I'm not missing something completely, the file is in the folder, but not included in NSB.Core project!

**Comment by timbussmann** [comment]
> how did you find that?

tbh, that was @andreasohlund finding I just had to play the messenger but I'll happily take the credits for that ;) :P

> Because if I'm not missing something completely, the file is in the folder, but not included in NSB.Core project!

I think you're missing something completely https://github.com/Particular/NServiceBus/blob/c3fcb674c9f0d634a19eb857665fb44fc67d5215/src/NServiceBus.Core/NServiceBus.Core.csproj#L122 ;)

**Pull request merged**